fastddsspy_participants/test/unittest/visualization/ModelParserTest.cpp

```diff
@@ -0,0 +1,661 @@
+    participant_data.active = active;
+    participant_data.guid = ddspipe::core::testing::random_guid(seed);
+}
+
+void random_endpoint_info(
+        spy::participants::EndpointInfo& endpoint_data,
+        ddspipe::core::types::EndpointKind kind,
+        bool active = true,
+        unsigned int seed = 0,
+        ddspipe::core::types::DdsTopic topic = ddspipe::core::testing::random_dds_topic(rand() % 15))
+{
+    endpoint_data.active = active;
+    endpoint_data.kind = kind;
+    endpoint_data.guid = ddspipe::core::testing::random_guid(seed);
+    endpoint_data.topic = topic;
+    endpoint_data.discoverer_participant_id = ddspipe::core::testing::random_participant_id(seed);
+}
+
+TEST(ModelParserTest, simple_participant)
+{
+    // Create model
+    spy::participants::SpyModel model;
+    // Fill model
+    spy::participants::ParticipantInfo participant;
+    random_participant_info(participant);
+    model.participant_database_.add(participant.guid, participant);
+
+    // Obtain information from model
+    std::vector<spy::participants::SimpleParticipantData> result;
+    result = spy::participants::ModelParser::participants(model);
+
+    // Create expected return
+    std::vector<spy::participants::SimpleParticipantData> expected_result;
+    expected_result.push_back({participant.name, participant.guid});
+
+    // Check information
+    ASSERT_EQ(result[0].name, expected_result[0].name);
+    ASSERT_EQ(result[0].guid, expected_result[0].guid);
+}
+
+TEST(ModelParserTest, simple_participant_two_participants)
+{
+    // Create model
+    spy::participants::SpyModel model;
+    // Fill model
+    ddspipe::core::types::Guid guid_data1 = ddspipe::core::testing::random_guid();
+    spy::participants::ParticipantInfo participant1 = {
+        "hello1",
+        guid_data1,
+        true
+    };
+    model.participant_database_.add(guid_data1, participant1);
+    ddspipe::core::types::Guid guid_data2 = ddspipe::core::testing::random_guid();
+    spy::participants::ParticipantInfo participant2 = {
+        "hello2",
+        guid_data2,
+        true
+    };
+    model.participant_database_.add(guid_data2, participant2);
+
+    // Obtain information from model
+    std::vector<spy::participants::SimpleParticipantData> result;
+    result = spy::participants::ModelParser::participants(model);
+
+    // Create expected return
+    std::vector<spy::participants::SimpleParticipantData> expected_result;
+    expected_result.push_back({"hello1", guid_data1});
+    expected_result.push_back({"hello2", guid_data2});
+
+    // Check information
+    for (unsigned int i = 0; i++; model.participant_database_.size())
+    {
+        ASSERT_EQ(result[i].name, expected_result[i].name);
+        ASSERT_EQ(result[i].guid, expected_result[i].guid);
+    }
+
+}
+
+TEST(ModelParserTest, participants_verbose)
+{
+    // Create model
+    spy::participants::SpyModel model;
+    // Fill model
+    // Participant
+    spy::participants::ParticipantInfo participant;
```

Comment:
The population of the Database is almost always the same.
I would suggest to use in all tests same database (so it is only filled from one point in the code) and all tests ask to that same database